Content.Shared/Damage/Components/DamageableComponent.cs

```diff
@@ -433,19 +419,75 @@ public void OnExplosion(ExplosionEventArgs eventArgs)
                 _ => throw new ArgumentOutOfRangeException()
             };
 
-            ChangeDamage(GetDamageType("Piercing"), damage, false);
-            ChangeDamage(GetDamageType("Heat"), damage, false);
+            foreach (var typeID in ExplosionDamageTypeIDs)
+            {
+                TryChangeDamage(_prototypeManager.Index<DamageTypePrototype>(typeID), damage);
+            }
+        }
+
+        /// <summary>
+        ///     Take a dictionary with <see cref="IPrototype"/> keys and return a dictionary using <see cref="IPrototype.ID"/> as keys
+        ///     instead.
+        /// </summary>
+        /// <remarks>
+        ///     Useful when sending damage type and group prototypes dictionaries over the network.
+        /// </remarks>
+        public static IReadOnlyDictionary<string, int>
+            ConvertDictKeysToIDs<TPrototype>(IReadOnlyDictionary<TPrototype, int> prototypeDict)
+            where TPrototype : IPrototype
+        {
```

Comment:
cant the parameter just be `IReadOnlyDictionary<IPrototype, Int>`?

Reply:
Just doing that results in an error about not being able to convert from `IReadOnlyDictionary<DamageTypePrototype, Int>` to `IReadOnlyDictionary<IPrototype, Int>`. There might be a way of simplifying it, but maybe I just dont know enough about c# & generics